docs/source/command_line.rst

```diff
@@ -482,6 +482,32 @@ of the above sections.
            # 'items' now has type List[List[str]]
            ...
 
+.. option:: --local-partial-types
+
+    By default, mypy won't check partial types spanning module top level or class top level.
+    This flag changes the behavior to only allow partial types at local level. The most common
+    cases are variables initialized with values such as ``[]``, ``{}`` where the explicit type
```

Comment:
This option only affects partial `None` types, and discussing `[]`, `{}`, etc. could be confusing here, since they aren't affected by the option.